--- A/CMakeLists.txt	Tue Aug 11 16:06:08 2020
+++ B/CMakeLists.txt	Tue Aug 11 16:21:29 2020
@@ -802,21 +802,21 @@
             LIBRARY DESTINATION "${INSTALL_LIB_DIR}")
     install(EXPORT ${PROJECT_NAME}
             DESTINATION "${INSTALL_CMAKE_DIR}"
-            NAMESPACE "MINIZIP::")
+            NAMESPACE "MINIZIPFORK::")
 
     # Create and install CMake package config version file to allow find_package()
     include(CMakePackageConfigHelpers)
-    write_basic_package_version_file(${CMAKE_CURRENT_BINARY_DIR}/minizipConfigVersion.cmake
+    write_basic_package_version_file(${CMAKE_CURRENT_BINARY_DIR}/minizipforkConfigVersion.cmake
         COMPATIBILITY SameMajorVersion)
 
-    file(WRITE minizipConfig.cmake.in "@PACKAGE_INIT@")
+    file(WRITE minizipforkConfig.cmake.in "@PACKAGE_INIT@")
 
     # Create config for find_package()
-    configure_package_config_file(minizipConfig.cmake.in minizipConfig.cmake
+    configure_package_config_file(minizipforkConfig.cmake.in minizipforkConfig.cmake
         INSTALL_DESTINATION "${INSTALL_CMAKE_DIR}")
 
-    install(FILES ${CMAKE_CURRENT_BINARY_DIR}/minizipConfigVersion.cmake
-                  ${CMAKE_CURRENT_BINARY_DIR}/minizipConfig.cmake
+    install(FILES ${CMAKE_CURRENT_BINARY_DIR}/minizipforkConfigVersion.cmake
+                  ${CMAKE_CURRENT_BINARY_DIR}/minizipforkConfig.cmake
             DESTINATION "${INSTALL_CMAKE_DIR}")
 endif()
 if(NOT SKIP_INSTALL_HEADERS AND NOT SKIP_INSTALL_ALL)
